### ISSUE TITLE
iOS: Terminated app receives launch notification twice

### ISSUE THREAD

**Issue by Bungeefan** [ben-xD/push#24]
When the app receives a push notification after it was force-quitted, the same notification is delivered twice.
Once via `notificationTapWhichLaunchedAppFromTerminated` **and** again when listening on `onNotificationTap`.

Used code (simplified):
```dart
void main() async {
  Map<String?, Object?>? data = await Push.instance.notificationTapWhichLaunchedAppFromTerminated;
  runApp(app);
}

class Main extends StatefulWidget {
  void initState() {
    Push.instance.onNotificationTap.listen((data) {
      // Same data as above
    }
  }
}
```

Reproduction steps:
* Force quit your app (Swipe up from recent apps)
* Trigger remote notification.
* Tap notification, which launches the app.
* App can access the same notification via `notificationTapWhichLaunchedAppFromTerminated` and receives it again in `onNotificationTap`.

Is this intended behavior?
This is not reproducible on Android.

**Comment by ben-xD** [comment]
Hey @Bungeefan, thanks for detailed report.

I've reproduced it and fixed it in https://github.com/ben-xD/push/pull/25, which is available as push `1.0.3`.

**Comment by Bungeefan** [comment]
I was curious and wanted to find out what might have happened and did a little research:

Looking at the [events of this repo](https://api.github.com/repos/ben-xD/push/events), there is a PushEvent after your Merge-PushEvent whose `head` doesn't contain the `before` commit as ancestor. [(Which apparently indicates a force-push)](https://stackoverflow.com/questions/17503226/find-committer-of-a-force-push-on-github#comment25444966_17503259)
Even GitHub says that the merged commit from main no longer belongs to a branch: https://github.com/ben-xD/push/commit/324922adb6507c1b1c38092f81ece29456551597

My theory would be that you accidentally force-pushed to main without pulling the previously merged commit locally and, therefore, overwriting the via GitHub merged PR commit.

**Comment by ben-xD** [comment]
Oh sorry! Thanks for investigating @Bungeefan. Sorry for the hassle. I tried to rush it given my time constraints a few days ago and this is the consequence.

It's now available in push `1.0.4`

**Comment by Bungeefan** [comment]
No problem, again, thanks for your quick action!
This seems indeed to be working with `push: 1.0.4`.
Thank you very much!
